src/components/ColumnPicker/NewColumPickerCategories.tsx

```diff
@@ -0,0 +1,159 @@
+import { CircularProgress } from "@material-ui/core";
+import {
+  Box,
+  Button,
+  Checkbox,
+  List,
+  SearchInput,
+  Text,
+} from "@saleor/macaw-ui/next";
+import React from "react";
+import { FormattedMessage, useIntl } from "react-intl";
+
+import messages from "./messages";
+import { NewColumnPickerPagination } from "./NewColumnPickerPagination";
+import { ColumnCategory } from "./useColumns";
+import { getExitIcon, getExitOnClick } from "./utils";
+
+export interface NewColumnPickerCategoriesProps {
+  columnCategories: ColumnCategory[];
+  columnPickerSettings: string[];
+  onClose: () => void;
+  onDynamicColumnSelect: (columns: string[]) => void;
+}
+
+export const NewColumnPickerCategories: React.FC<
+  NewColumnPickerCategoriesProps
+> = ({
+  columnCategories,
+  onClose,
+  onDynamicColumnSelect,
+  columnPickerSettings,
+}) => {
+  const intl = useIntl();
+
+  const [selectedCategory, setSelectedCategory] = React.useState<string>();
+  const [query, setQuery] = React.useState<string>("");
+
+  const currentCategory = React.useMemo(
+    () => columnCategories.find(category => category.name === selectedCategory),
+    [columnCategories, selectedCategory],
+  );
+
+  const changeHandler = (column: string) => {
+    if (columnPickerSettings.includes(column)) {
+      onDynamicColumnSelect(
+        columnPickerSettings.filter(currentCol => currentCol !== column),
+      );
+    } else {
+      onDynamicColumnSelect([...columnPickerSettings, column]);
+    }
+  };
+
+  React.useEffect(() => {
+    // Preselect category when there is only one
+    if (columnCategories.length === 1) {
+      setSelectedCategory(columnCategories[0].name);
+    }
+  }, [columnCategories]);
```

Comment:
Column categories come from API so they might be undefined at first, I haven't implemented it with strict nulls in mind. If I set this as initial value of useState it might not work when component is rendered without data from the API